File: data/multilang_comparison/program_julia/src/func18.c
#include "program_julia.h" 
array_t* func18(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array157 = func19(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array157->id);
      free(params0.data);
      array157->refC--;
      if(array157->refC == 0) {
         free(array157->data);
         free(array157);
         DEBUG_FREE(array157->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array167 = func20(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array167->id);
      free(params0.data);
      array167->refC--;
      if(array167->refC == 0) {
         free(array167->data);
         free(array167);
         DEBUG_FREE(array167->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array172 = func21(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array172->id);
   free(params0.data);
   return array172;
}

File: data/multilang_comparison/program_julia/src/func18.jl
include("program_julia_head.jl")
function func18(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array125 = func19(params0, get_path(), loopsFactor);
      array125.refC -= 1
      if array125.refC == 0
          empty!(array125.data)
          array125 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array135 = func20(params0, get_path(), loopsFactor);
      array135.refC -= 1
      if array135.refC == 0
          empty!(array135.data)
          array135 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array140 = func21(params0, get_path(), loopsFactor);
   return array140;
end

